src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/fieldTransformers/IconFieldTransformer.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import React from 'react';
+import type {Node} from 'react';
+import log from 'loglevel';
+import Icon from '../../../components/Icon';
+import type {FieldTransformer} from '../types';
+import iconFieldTransformerStyles from './iconFieldTransformer.scss';
+
+export default class IconFieldTransformer implements FieldTransformer {
+    transform(value: *, parameters: {[string]: mixed}): Node {
+        if (!value) {
+            return value;
+        }
+
+        const {mapping} = parameters;
+        if (!mapping) {
+            return value;
+        }
+
+        if (typeof mapping !== 'object') {
+            log.error('Transformer parameter "mapping" needs to be of type collection.');
+
+            return null;
+        }
+
+        const iconConfig = mapping[value];
+        if (!iconConfig) {
+            log.warn(`Transformer parameter "mapping/${value}" is not set.`);
+
+            return value;
+        }
+
+        if (typeof iconConfig === 'object') {
+            return this.transformObjectConfig(value, iconConfig);
+        }
+
+        if (typeof iconConfig === 'string') {
+            return this.transformStringConfig(iconConfig);
+        }
+
+        log.error(`Transformer parameter "mapping/${value}" needs to be either of type string or collection.`);
+
+        return null;
+    }
+
+    transformObjectConfig(value: *, iconConfig: Object): Node {
+        if (!iconConfig.icon) {
+            log.error(`Transformer parameter "mapping/${value}/icon" needs to be set.`);
+
+            return null;
+        }
+
+        if (typeof iconConfig.icon !== 'string') {
+            log.error(`Transformer parameter "mapping/${value}/icon" needs to be of type string.`);
+
+            return null;
+        }
+
+        const icon = iconConfig.icon;
+
+        if (iconConfig.color) {
+            if (typeof iconConfig.color !== 'string') {
```

Comment:
With the above in place we could write the check like this:

```javascript
if (color !== undefined && typeof color !== 'string') {
    log.error('...');

    return null;
}
```

Then the style could be created like this, and we could avoid creating the same jsx twice:

```
const style = {};

if (color) {
    style.color = color;
}

return <Icon style={style} />;
```